uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -247,9 +264,56 @@ define([
          * @param {string} key - The property name
          */
         toAPIFilter: function (result, __, key) {
-          if (!_.includes(['balance_change', 'dates', 'comments'], key)) {
+          if (!_.includes(['balance_change', 'dates', 'comments', 'uploader', 'files'], key)) {
             result[key] = this[key];
           }
+        },
+
+        /**
+        * Gets file attachments associated with this leave request
+        *
+        * @return {Promise}
+        */
+        getAttachments: function () {
+          return LeaveRequestAPI.getAttachments(this.id)
+            .then(function (attachments) {
+              this.files = attachments;
+            }.bind(this));
+        },
+
+        /**
+        * Deletes the given attachment from local files array. It sets a flag
+        * to delete in final update call. This can be used by the client to set
+        * file for deletion.
+        */
+        deleteAttachment: function (attachmentID) {
```

Comment:
The name of the method (`delete..`) and the documentation ("Deletes the given attachment from local files array") are not describing what is actually going on in the method itself. You are not deleting anything, you are just marking a file to be deleted later. So which one is it? Should you mark the file for deletion, or delete it from the array?

Shouldn't this be similar to how comments are deleted? Can you talk with @deb1990 about it since he wrote the comments bits?